src/avram/add_column_attributes.cr

```diff
@@ -0,0 +1,125 @@
+# This module creates methods for each column in a model that map
+# to an `Avram::Attribute` as well as methods that fill those attributes
+# with values that comes from params.
+module Avram::AddColumnAttributes
+  # :nodoc:
+  macro add_column_attributes(attributes)
+    {% for attribute in attributes %}
+      {% COLUMN_ATTRIBUTES << attribute %}
+    {% end %}
+
+    private def extract_changes_from_params
+      permitted_params.each do |key, value|
+        {% for attribute in attributes %}
+          {% if attribute[:type].is_a?(Generic) %}
+            set_{{ attribute[:name] }}_from_param(value.as(Array(String))) if key == {{ attribute[:name].stringify }}
+          {% else %}
+            set_{{ attribute[:name] }}_from_param(value.as(String)) if key == {{ attribute[:name].stringify }}
```

Comment:
If we don't cast these here, then `typeof(value)` is `Array(String) | String`, and the compiler starts to get confused